### PR TITLE
Add multiarch build

### PR DESCRIPTION
**Pull request by lexfrei**
Important! Add CR_PAT (container registry personal access token) to repo secrets. Required permissions: packages:*

- Building on each push. Add the desired config, example: https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-syntax-for-github-actions#on
- Current tagging: commit sha and branch name (ex.: sha-8a83892, dev)
- Change image repository to the desired one at L:12
- GitHub Container registry used. Feel free to change it on L:27
- ARM64/AMD64 build enabled. Add more on L:36

### CODE REVIEW

**Comment by lexfrei** [comment]
Replacing: #1022

**Comment by Odianosen25** [comment]
Hello @lexfrei,

Thanks for this. Just moving via the PRs and wanted to ask a favour from yours.

There is a section of users who prefer a Debain based one (for good reason), compared to the Alpine which is default. Will it be possible please, to have a multi arch build for both?

What will it entail?

Kind regards

**Comment by lexfrei** [comment]
@Odianosen25 this CI is not about alpine-only. It refers to Dockerfile on the root of the project and just builds it. I can't create a proper Dockerfile for debiand-based images just coz I'm not similar to deb-world and cant suggest how to make a good one.

**Comment by Odianosen25** [comment]
Hello @lexfrei,

Thanks for the response, and I am sorry to drag this a little, but I am trying to find solution to something. There is a PR [here](https://github.com/AppDaemon/appdaemon/pull/1118), which seeks to make use of deb as base image, and I kind of understand if some users want a deb, due to it being easier.

From what you saying, seems you are an alpine person, which I am also, but I will like AD users to have both worlds if they can. So if you saying all your script does is make use of the existing docker file and build on that. Is it possible in anyway, to make use of the PR submitted to make both builds?

 Kind regards

**Comment by benleb** [comment]
> From what you saying, seems you are an alpine person, which I am also, but I will like AD users to have both worlds if they can. So if you saying all your script does is make use of the existing docker file and build on that. Is it possible in anyway, to make use of the PR submitted to make both builds?

Hey, I would suggest two separate Dockerfiles, `debian.Dockerfile` and `alpine.Dockerfile`, which will then be built by a github workflow. And yes, building both is no problem... in this case you would just have do duplicate the block in the yaml starting with "  build:" (below "jobs:") and specify the Dockerfiles in each of them. See you in discord :D

File: .github/workflows/docker.yml [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+        registry: ghcr.io
+        username: ${{ github.repository_owner }}
+        password: ${{ secrets.CR_PAT }}
+
+    - name: Build and push
+      uses: docker/build-push-action@v2
+      with:
+        context: .
+        file: ./Dockerfile
+        platforms: linux/amd64,linux/arm64
```
Comment (benleb) [inline .github/workflows/docker.yml:36]: lets also add `linux/arm/v7,linux/arm/v6` for compatibility with old Pis for example.

**Review by benleb — CHANGES_REQUESTED** [review]
A more up-to-date version of my PR (#1022) 👍  But I would suggest including the other archs as this has no downsides but makes it possible to use the images on old RPis. Should not make any problems, I am building appdaemon for it since months without issues 👍

**Comment by lexfrei** [comment]
@benleb added

**Review by benleb — APPROVED** [review]

**Comment by lexfrei** [comment]
Don't merge. Let me bump action versions

**Comment by Odianosen25** [comment]
Thanks for the work on this guys, really appreciate it. @lexfrei just let me know when you done then will look into merging sometime later today or tomorrow, once I get my head around it all.

kind regards

**Comment by lexfrei** [comment]
@Odianosen25 looks done for me.
Also, the registry was changed to the suggested one. Feel free to change it again

**Comment by lexfrei** [comment]
Also, I can suggest using depandabot for GH actions to keep actions updated.

Create `.github/dependabot.yml` with similar content:

```yaml
version: 2
updates:
- package-ecosystem: github-actions
  directory: "/"
  schedule:
    interval: daily
    time: "02:00"
  open-pull-requests-limit: 10
- package-ecosystem: pip
  directory: "/"
  schedule:
    interval: daily
    time: "02:00"
  open-pull-requests-limit: 10
- package-ecosystem: docker
  directory: "/"
  schedule:
    interval: daily
    time: "02:00"
  open-pull-requests-limit: 10
```

Or add GH Actions to the job list on dependabot site.

**Comment by benleb** [comment]
As that's completely independent it should get its own PR 👍

**Comment by lexfrei** [comment]
@benleb just wild guess :) I got no idea how dependabot configured right now for the project, but it needs adjustment to update the action from this PR

**Comment by lexfrei** [comment]
@Odianosen25 add token as CR_PAT in repo config and rerun job

**Comment by Odianosen25** [comment]
@lexfrei,

Was just reminded abt that, will be worked on as I need to discuss it with Andrew on how he wants the secret to be created.

I am sure he will look into it once he comes on line.

Regards

**Comment by nicolaerosia** [comment]
is the CI pushing arm64 images now? If yes, where?

**Comment by Odianosen25** [comment]
No @nicolaerosia, unfortunately its still failing but @acockburn is looking into it.

Regards

**Comment by Enf0** [comment]
Any updates on this? Is there an arm64 image of AppDaemon available?

**Comment by benleb** [comment]
until then, you can use mine if you want, stripped down and arm64 available ✌️

https://github.com/benleb/appdaemon
https://hub.docker.com/r/benleb/appdaemon

**Comment by emilp333** [comment]
Thanks! I will try it :)

**Pull request merged**